Supervised, interactive genomic data analysis

This slide deck is a summary of my contributions to supervised and
interactive data analysis in the past few years (2011-2014).

The most interesting parts are the new xypics that clearly show the
differences between supervised (plot before modeling) and unsupervised
analysis (plot after modeling).

The other main contribution is the 2 x 2 table that classifies
the types of analysis methods:

| Evaluation:         | qualitative               | quantitative                        |
| Input: data $x$ +   | parameters                | labels $y$                          |
|                     |                           |                                     |
|                     | unsupervised              | supervised                          |
|---------------------+---------------------------+-------------------------------------|
| *non-interactive*   | default parameters        | trained parameters                  |
| (fit one model)     | plot $x$                  | plot $x, y$                         |
|                     | e.g. [[http://onlinelibrary.wiley.com/doi/10.1111/j.1541-0420.2006.00662.x/full][cghseg + mBIC]]        | e.g. [[http://jmlr.org/proceedings/papers/v28/hocking13.html'][Max Margin interval regression]] |
|---------------------+---------------------------+-------------------------------------|
| *interactive*       | manually tuned parameters | iteratively trained parameters      |
| (plot/update model) | plot $x, f(x, \lambda)$   | plot $x, y, f(x, \lambda)$          |
|                     | e.g. [[http://compbio.med.harvard.edu/CGHweb/][CGHweb]]               | e.g. [[http://bioinformatics.oxfordjournals.org/content/30/11/1539][SegAnnDB]]                       |


(original version first appeared https://bitbucket.org/tdhock/intro-slides/src)

Finally, to get the R source code and data for the result
figures/tables, you have to look at my original papers:
- [[http://www.biomedcentral.com/1471-2105/14/164][BMC Bioinformatics 2013 paper]]. [[http://cran.r-project.org/web/packages/bams/vignettes/redoing-the-article.pdf][Code for "Learning smoothing models..." paper in bams package on CRAN]]
- [[http://jmlr.org/proceedings/papers/v28/hocking13.html][ICML 2013 paper]]. [[https://gforge.inria.fr/scm/viewvc.php/pruned-dp/article/?root%3Dbreakpoints][Code for "Learning sparse penalties..." paper]].
- [[http://bioinformatics.oxfordjournals.org/content/30/11/1539][Bioinformatics 2014 paper]]. [[https://gforge.inria.fr/scm/viewvc.php/webapp/applications-note/?root%3Dbreakpoints][Code for SegAnnDB paper]].
- PeakSeg, NIPS workshop on machine learning in computational biology 2014. [[https://github.com/tdhock/PeakSegDP-NIPS][Code]] for [[https://github.com/tdhock/PeakSegDP-NIPS/blob/master/HOCKING-RIGAILL-PeakSegDP-NIPS-poster.pdf?raw%3Dtrue][poster]] and [[https://github.com/tdhock/PeakSegDP-NIPS/blob/master/HOCKING-RIGAILL-chip-seq-NIPS.pdf?raw%3Dtrue][4-page paper]].
